### PR TITLE
This commit abstracts some functionality to facilitate testing.

### PR DESCRIPTION
**Pull request by spacejam**
- Abstract calls to Exit to facilitate testing.
- Abstract the pod status function for testing purposes.
- Fix argument ordering of Prefix checking.
- Address @sttts's feedback.

### CODE REVIEW

**Comment by sttts** [comment]
A last little nit: the dot at the end of the commit msg usually is not put there.

File: plugin/contrib/mesos/pkg/executor/executor.go [hunk 1]
```diff
@@ -701,7 +706,10 @@ func (k *KubernetesExecutor) Shutdown(driver bindings.ExecutorDriver) {
 // assumes that caller has obtained state lock
 func (k *KubernetesExecutor) doShutdown(driver bindings.ExecutorDriver) {
 	defer func() {
-		log.Exit("exiting with unclean shutdown: %v", recover())
+		log.Warningf("exiting with unclean shutdown: %v", recover())
```
Comment (jdef) [inline plugin/contrib/mesos/pkg/executor/executor.go]: s/Warningf/Errorf/

File: plugin/contrib/mesos/pkg/executor/service/service.go [hunk 2]
```diff
@@ -360,6 +360,10 @@ func (ks *KubeletExecutorServer) createAndInitKubelet(
 				}
 			}
 		},
+		ExitFunc: func(code int) { os.Exit(code) },
```
Comment (jdef) [inline plugin/contrib/mesos/pkg/executor/service/service.go]: pretty sure you can write: `ExitFunc: os.Exit`

**Comment by jdef** [comment]
minor nits, otherwise LGTM

File: plugin/contrib/mesos/pkg/executor/executor.go [hunk 3]
```diff
@@ -675,7 +680,7 @@ func (k *KubernetesExecutor) FrameworkMessage(driver bindings.ExecutorDriver, me
 
 	log.Infof("Receives message from framework %v\n", message)
 	//TODO(jdef) master reported a lost task, reconcile this! @see scheduler.go:handleTaskLost
-	if strings.HasPrefix("task-lost:", message) && len(message) > 10 {
+	if strings.HasPrefix(message, "task-lost:") && len(message) > 10 {
```
Comment (jdef) [inline plugin/contrib/mesos/pkg/executor/executor.go:683]: nice catch!

**Comment by jdef** [comment]
:ship:

**Pull request merged**